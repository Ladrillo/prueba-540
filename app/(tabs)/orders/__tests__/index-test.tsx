import { render, waitFor } from '@testing-library/react-native'
import { ApolloProvider } from '@apollo/client'
import client from '@/apollo/client'
import OrdersScreen from '../index'

describe('<OrdersScreen />', () => {
  let screen: any
  beforeEach(() => {
    screen = render(
      <ApolloProvider client={client}>
        <OrdersScreen />
      </ApolloProvider>
    )
  })
  test('Text renders correctly on OrdersScreen', async () => {
    await waitFor(() => {
      screen.getByText('Amazon')
      screen.getByText('Ebay')
      screen.getByText('Costco')
      // screen.debug()
    })
    expect(screen.toJSON()).toMatchSnapshot()
  })
})
