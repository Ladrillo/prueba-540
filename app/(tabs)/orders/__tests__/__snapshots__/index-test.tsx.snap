// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrdersScreen /> Text renders correctly on OrdersScreen 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "flex": 1,
      },
    ]
  }
>
  <RCTSafeAreaView
    style={
      {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          undefined,
          {
            "fontSize": 32,
            "fontWeight": "bold",
            "lineHeight": 32,
          },
          undefined,
          undefined,
          undefined,
          {
            "marginLeft": 15,
            "marginTop": 40,
          },
        ]
      }
    >
      Orders
    </Text>
    <RCTScrollView
      contentContainerStyle={
        {
          "paddingBottom": 80,
        }
      }
    >
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "#fff",
                "borderRadius": 15,
                "elevation": 5,
                "margin": 20,
                "marginBottom": 0,
                "overflow": "hidden",
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 5,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                {
                  "height": 120,
                  "justifyContent": "space-between",
                  "resizeMode": "cover",
                  "width": "100%",
                }
              }
            >
              <Image
                source={
                  {
                    "uri": "https://res.cloudinary.com/caskchain/image/upload/v1717158058/Sequra/amazon.png",
                  }
                }
                style={
                  [
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    {
                      "height": 120,
                      "width": "100%",
                    },
                    undefined,
                  ]
                }
              />
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "padding": 15,
                  }
                }
              >
                <View>
                  <Text
                    style={
                      [
                        {
                          "color": "#11181C",
                        },
                        {
                          "fontSize": 16,
                          "lineHeight": 24,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "color": "#fff",
                          "fontSize": 20,
                          "fontWeight": "bold",
                        },
                      ]
                    }
                  >
                    Amazon
                  </Text>
                  <View
                    style={
                      {
                        "flexDirection": "row",
                        "marginTop": 5,
                      }
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "#11181C",
                          },
                          {
                            "fontSize": 16,
                            "lineHeight": 24,
                          },
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {
                            "color": "#fff",
                            "marginRight": 10,
                          },
                        ]
                      }
                    >
                      1 nov 2021
                    </Text>
                    <Text
                      style={
                        [
                          {
                            "color": "#11181C",
                          },
                          {
                            "fontSize": 16,
                            "lineHeight": 24,
                          },
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {
                            "color": "#fff",
                          },
                        ]
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          [
                            {
                              "color": undefined,
                              "fontSize": 14,
                            },
                            undefined,
                            {
                              "fontFamily": "FontAwesome",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            {},
                          ]
                        }
                      >
                        
                      </Text>
                       
                      5
                       artículos
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#fff",
                      "borderRadius": 20,
                      "height": 40,
                      "justifyContent": "center",
                      "marginRight": 10,
                      "width": 40,
                    }
                  }
                >
                  <Image
                    source={
                      {
                        "uri": "https://cdn.iconscout.com/icon/free/png-512/free-amazon-1543560-1306063.png",
                      }
                    }
                    style={
                      {
                        "borderRadius": 17.5,
                        "height": 35,
                        "width": 35,
                      }
                    }
                  />
                </View>
              </View>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "padding": 15,
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#e0f7fa",
                    "borderRadius": 10,
                    "height": 110,
                    "justifyContent": "center",
                    "paddingHorizontal": 15,
                    "paddingVertical": 10,
                    "width": 110,
                  }
                }
              >
                <Text
                  style={
                    {
                      "fontSize": 24,
                      "fontWeight": "bold",
                    }
                  }
                >
                  1
                </Text>
                <Text
                  style={
                    {
                      "fontSize": 14,
                    }
                  }
                >
                  noviembre
                </Text>
              </View>
              <View
                style={
                  {
                    "alignItems": "flex-end",
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#11181C",
                      },
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "color": "#7e7e7e",
                        "fontSize": 14,
                      },
                    ]
                  }
                >
                  Cobro de:
                </Text>
                <Text
                  style={
                    [
                      {
                        "color": "#11181C",
                      },
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "color": "#000",
                        "fontSize": 22,
                        "fontWeight": "bold",
                      },
                    ]
                  }
                >
                  100
                   €
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "#fff",
                "borderRadius": 15,
                "elevation": 5,
                "margin": 20,
                "marginBottom": 0,
                "overflow": "hidden",
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 5,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                {
                  "height": 120,
                  "justifyContent": "space-between",
                  "resizeMode": "cover",
                  "width": "100%",
                }
              }
            >
              <Image
                source={
                  {
                    "uri": "https://res.cloudinary.com/caskchain/image/upload/v1717158059/Sequra/ebay.png",
                  }
                }
                style={
                  [
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    {
                      "height": 120,
                      "width": "100%",
                    },
                    undefined,
                  ]
                }
              />
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "padding": 15,
                  }
                }
              >
                <View>
                  <Text
                    style={
                      [
                        {
                          "color": "#11181C",
                        },
                        {
                          "fontSize": 16,
                          "lineHeight": 24,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "color": "#fff",
                          "fontSize": 20,
                          "fontWeight": "bold",
                        },
                      ]
                    }
                  >
                    Ebay
                  </Text>
                  <View
                    style={
                      {
                        "flexDirection": "row",
                        "marginTop": 5,
                      }
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "#11181C",
                          },
                          {
                            "fontSize": 16,
                            "lineHeight": 24,
                          },
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {
                            "color": "#fff",
                            "marginRight": 10,
                          },
                        ]
                      }
                    >
                      1 nov 2021
                    </Text>
                    <Text
                      style={
                        [
                          {
                            "color": "#11181C",
                          },
                          {
                            "fontSize": 16,
                            "lineHeight": 24,
                          },
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {
                            "color": "#fff",
                          },
                        ]
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          [
                            {
                              "color": undefined,
                              "fontSize": 14,
                            },
                            undefined,
                            {
                              "fontFamily": "FontAwesome",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            {},
                          ]
                        }
                      >
                        
                      </Text>
                       
                      10
                       artículos
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#fff",
                      "borderRadius": 20,
                      "height": 40,
                      "justifyContent": "center",
                      "marginRight": 10,
                      "width": 40,
                    }
                  }
                >
                  <Image
                    source={
                      {
                        "uri": "https://cdn.iconscout.com/icon/free/png-512/free-ebay-13-675708.png",
                      }
                    }
                    style={
                      {
                        "borderRadius": 17.5,
                        "height": 35,
                        "width": 35,
                      }
                    }
                  />
                </View>
              </View>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "padding": 15,
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#e0f7fa",
                    "borderRadius": 10,
                    "height": 110,
                    "justifyContent": "center",
                    "paddingHorizontal": 15,
                    "paddingVertical": 10,
                    "width": 110,
                  }
                }
              >
                <Text
                  style={
                    {
                      "fontSize": 24,
                      "fontWeight": "bold",
                    }
                  }
                >
                  1
                </Text>
                <Text
                  style={
                    {
                      "fontSize": 14,
                    }
                  }
                >
                  noviembre
                </Text>
              </View>
              <View
                style={
                  {
                    "alignItems": "flex-end",
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#11181C",
                      },
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "color": "#7e7e7e",
                        "fontSize": 14,
                      },
                    ]
                  }
                >
                  Cobro de:
                </Text>
                <Text
                  style={
                    [
                      {
                        "color": "#11181C",
                      },
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "color": "#000",
                        "fontSize": 22,
                        "fontWeight": "bold",
                      },
                    ]
                  }
                >
                  200
                   €
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "#fff",
                "borderRadius": 15,
                "elevation": 5,
                "margin": 20,
                "marginBottom": 0,
                "overflow": "hidden",
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 5,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                {
                  "height": 120,
                  "justifyContent": "space-between",
                  "resizeMode": "cover",
                  "width": "100%",
                }
              }
            >
              <Image
                source={
                  {
                    "uri": "https://res.cloudinary.com/caskchain/image/upload/v1717158062/Sequra/costco.png",
                  }
                }
                style={
                  [
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    {
                      "height": 120,
                      "width": "100%",
                    },
                    undefined,
                  ]
                }
              />
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "padding": 15,
                  }
                }
              >
                <View>
                  <Text
                    style={
                      [
                        {
                          "color": "#11181C",
                        },
                        {
                          "fontSize": 16,
                          "lineHeight": 24,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "color": "#fff",
                          "fontSize": 20,
                          "fontWeight": "bold",
                        },
                      ]
                    }
                  >
                    Costco
                  </Text>
                  <View
                    style={
                      {
                        "flexDirection": "row",
                        "marginTop": 5,
                      }
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "#11181C",
                          },
                          {
                            "fontSize": 16,
                            "lineHeight": 24,
                          },
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {
                            "color": "#fff",
                            "marginRight": 10,
                          },
                        ]
                      }
                    >
                      1 nov 2024
                    </Text>
                    <Text
                      style={
                        [
                          {
                            "color": "#11181C",
                          },
                          {
                            "fontSize": 16,
                            "lineHeight": 24,
                          },
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {
                            "color": "#fff",
                          },
                        ]
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          [
                            {
                              "color": undefined,
                              "fontSize": 14,
                            },
                            undefined,
                            {
                              "fontFamily": "FontAwesome",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            {},
                          ]
                        }
                      >
                        
                      </Text>
                       
                      15
                       artículos
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#fff",
                      "borderRadius": 20,
                      "height": 40,
                      "justifyContent": "center",
                      "marginRight": 10,
                      "width": 40,
                    }
                  }
                >
                  <Image
                    source={
                      {
                        "uri": "https://cdn.iconscout.com/icon/free/png-512/free-costco-282448.png",
                      }
                    }
                    style={
                      {
                        "borderRadius": 17.5,
                        "height": 35,
                        "width": 35,
                      }
                    }
                  />
                </View>
              </View>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "padding": 15,
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#e0f7fa",
                    "borderRadius": 10,
                    "height": 110,
                    "justifyContent": "center",
                    "paddingHorizontal": 15,
                    "paddingVertical": 10,
                    "width": 110,
                  }
                }
              >
                <Text
                  style={
                    {
                      "fontSize": 24,
                      "fontWeight": "bold",
                    }
                  }
                >
                  1
                </Text>
                <Text
                  style={
                    {
                      "fontSize": 14,
                    }
                  }
                >
                  noviembre
                </Text>
              </View>
              <View
                style={
                  {
                    "alignItems": "flex-end",
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#11181C",
                      },
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "color": "#7e7e7e",
                        "fontSize": 14,
                      },
                    ]
                  }
                >
                  Cobro de:
                </Text>
                <Text
                  style={
                    [
                      {
                        "color": "#11181C",
                      },
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "color": "#000",
                        "fontSize": 22,
                        "fontWeight": "bold",
                      },
                    ]
                  }
                >
                  300
                   €
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
</View>
`;
